QNN Llama Runner implement IRunner

rohansjoshi opened the pull request:
Summary:
This PR makes the Runner for running Qualcomm LlamaModels implement the IRunner interface

Using this, enable running static Llama models inside LlamaDemo Android app

Switched default eval mode to hybrid everywhere

Differential Revision: D79759817

shewu-quic reviewed (approved):
LGTM. Thank you for the efforts. It is really awesome 👍

cccclai reviewed (approved):
Looks good to me, also stamp on behalf of qcom team

rohansjoshi committed:
Rebased on top of recent changes

Reviewed By: cccclai

rohansjoshi committed:
QNN Llama Runner implement IRunner (#13171)

Summary:

This PR makes the Runner for running Qualcomm LlamaModels implement the IRunner interface

Using this, enable running static Llama models inside LlamaDemo Android app

Switched default eval mode to hybrid everywhere

Reviewed By: cccclai

Differential Revision: D79759817

rohansjoshi committed:
Fixed lint issues

rohansjoshi commented on examples/qualcomm/oss_scripts/llama/runner/runner.cpp:
```diff
@@ -9,6 +9,7 @@
 // A llama 3.2 runner that includes preprocessing and post processing
 // logic. The module takes in a string as input and emits a string as output.
 
+#include <executorch/examples/models/llama/runner/runner.h>
```
I'm only using the function `load_llama_tokenizer` from executorch/examples/models/llama/runner/runner, not the runner there. I'm trying to share code from examples/models
